### ISSUE TITLE
Blog related links with Attachments usable

### ISSUE THREAD

**Issue by strausmann** [squidfunk/mkdocs-material#4637]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I want to suggest an idea and checked that ...

- [X] ... to my best knowledge, my idea wouldn't break something for other users
- [X] ... the documentation does not mention anything about my idea
- [x] ... there are no open or closed issues that are related to my idea

### Description

Currently it seems that only external links and .md files can be linked to in Related Links. We would also be able to provide e.g. PDF, Word or other files via the Related Links.

### Use Cases

In the blog, it would be generally useful to place attached documents present by means of the Related Links. In texts with sometimes 5-6 A4 pages, the links to attachments would simply get lost.

We have already tried to link the attachments in the documents, then to link them via the assets URI in the Related Links. But this is quite complicated.

**Comment by squidfunk** [comment]
Thanks for reporting. Please provide some more information of how you imagine to use this feature.

**Comment by strausmann** [comment]
We use MkDocs Material Insider for an internal system for Company Information.

The posts are called Circulars and can have attachments. These attachments can be linked in the blog area of MkDocs Material currently only in the content of the post.

With Related Links we cannot link the documents, here only external links or other pages from MkDocs are possible.

![image](https://user-images.githubusercontent.com/4519228/201706724-09fe1062-8f45-4af1-83c8-521a3a0ad454.png)

So we have the following requirement. In a blog post files like PDF, Word, Excel etc. should be linked. Files can be included via the Related Links. The attachments are located in the project directly next to the post file (*.md) itself.

**Comment by strausmann** [comment]
Ahso, because with an attachment probably no meaningful name for the Related Link is possible. Currently there would always be "None". Would be a similar integration as required for external links.

![image](https://user-images.githubusercontent.com/4519228/201708953-eea963b1-cead-4674-a1ef-8d0235a61b40.png)

**Comment by squidfunk** [comment]
I'm not sure what you're doing, but I can't reproduce the problem – linking files works perfectly for me. For example, take the following configuration:

``` yaml
links:
  - setup/setting-up-site-search.md#built-in-search-plugin
  - insiders/index.md#how-to-become-a-sponsor
  - PDF: blog/posts/test.pdf
```

Results in:

<img width="233" alt="Bildschirmfoto 2022-11-15 um 19 24 42" src="https://user-images.githubusercontent.com/932156/201996986-7b6c8b5a-1b3d-4529-9d54-f705a7b8d7b9.png">

Clicking on the PDF opens it. I suspect that the path is wrong – the path __must__ start at the root of your `docs_dir`.

**Comment by squidfunk** [comment]
Okay, so the case where you don't specify the title for the PDF doesn't work correctly. That wasn't apparent for me from what you wrote. I'll reopen the issue and investigate.

**Comment by squidfunk** [comment]
Fixed in 606c88cc3. Non-Markdown links now correctly resolve to the assets destination paths after copying.

**Comment by squidfunk** [comment]
Released as part of [8.5.10+insiders-4.26.3](https://github.com/squidfunk/mkdocs-material-insiders/releases/tag/8.5.10-insiders-4.26.3).

**Comment by strausmann** [comment]
Thank you
